lib/screens/home_screen.dart

```diff
@@ -16,25 +22,114 @@ class HomeScreen extends StatefulWidget {
 }
 
 class _HomeScreenState extends State<HomeScreen> {
-  Future<List<StoryInfo>> getStories(int page, StoryInfo? last) async {
-    return await algolia.instance
-        .index('stories')
-        .query('')
-        .setPage(page)
-        // .filters('language:${settings.targetLanguage}')
-        .getObjects()
-        .then(
-          (s) => s.hits.map((h) => StoryInfo.fromAlgoliaHit(h)).toList(),
+  Future<List<StoryInfo>> getStories(
+    String index, {
+    int? page,
+    int? hitsPerPage,
+  }) async {
+    var query = algolia.instance.index(index).query('');
+    if (page != null) query = query.setPage(page);
+    if (hitsPerPage != null) query = query.setHitsPerPage(hitsPerPage);
+    final hits = await query.getObjects().then((s) => s.hits);
+    return hits.map((h) => StoryInfo.fromAlgoliaHit(h)).toList();
+  }
+
+  Widget categoryCards(IconData icon, String title, String index) {
+    return Column(
+      children: [
+        ListTile(
+          leading: Icon(icon),
+          horizontalTitleGap: 0,
+          title: Text(
+            capitalize(title),
+            style: Theme.of(context).textTheme.headline6,
+          ),
+          trailing: const Icon(Icons.arrow_forward_rounded),
+          onTap: () => categorySheet(title, index),
+        ),
+        LoadingBuilder(
+          future: getStories(index, hitsPerPage: 10),
+          builder: (context, stories) {
+            // TODO why is the type not inferenced?
+            final s = stories as List<StoryInfo>;
```

Comment:
Weird indeed, but can be fixed by supplying the generic type argument for `LoadingBuilder<List<StoryInfo>>`